### ISSUE TITLE
failed on macOS 10.13.6

### ISSUE THREAD

**Issue by jsvisa** [browsh-org/browsh#189]
Just a few points to consider before submitting a bug report:

  * Do a quick search for any existing issues describing your bug
 none
  * Give a clear and concise description of what the bug is
failed to start
  * Include the contents of your `./debug.log` generated with `browsh --debug`
errors in the terminal are: 
```   
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation  code=0x1 addr=0x0 pc=0x10b9c4e]

goroutine 13 [running]:
os.(*Process).signal( 0x0, 0x1597e40, 0x17b5378, 0x0, 0x0)
/home/travis/.gimme/versions/go1.10.3.linux.amd64/src/os/exec_unix.go :56 +0x2e
os.(*Process).Signal(0x0, 0x1597e40, 0x17b5378, 0xc42067de80 , 0x18)
/home/travis/.gimme/versions/go1.10.3.linux.amd64/src/os/exec.go:129 +0x3f
os.(*Process).kill(0x0, 0xc4206b0328, 0xc42067de80)
/home/travis/.gimme/versions/go1.10.3.linux.amd64/src/os/exec_posix.go:54 +0x43
os.(*Process).Kill(0x0, 0x4, 0x4)
/home/travis/.gimme/versions/go1.10.3.linux.amd64/src/os/exec.go:114 +0x2b
browsh/interfacer/src/browsh.startHeadlessFirefox()
/home/travis/gopath/src/browsh/interfacer/src/browsh/firefox.go:85 +0x3b2
created by browsh/interfacer/src/browsh.setupFirefox
/home/travis/gopath/src/browsh/interfacer/src/browsh/firefox.go:288 +0x35
~/Downloads$                                                                                                                                                               
```

and the debug.log looks like: 
```
Attempting to connect to Firefox Marionette
FF-MRNT: 50:{"applicationType":"gecko","marionetteProtocol":3}
Sending `newSession` to Firefox Marionette
FF-MRNT: 544:[1,0,null,{"sessionId":"693d1ccc-35c3-014e-8ffb-304e7b85b052","capabilities":{"browserName":"firefox","browserVersion":"62.0","platformName":"darwin","platformVersion":"17.7.0","pageLoadStrategy":"normal","acceptInsecureCerts":false,"timeouts":{"implicit":0,"pageLoad":300000,"script":30000},"rotatable":false,"moz:accessibilityChecks":false,"moz:headless":true,"moz:processID":60148,"moz:profile":"/Users/zhengdelweng/Library/Application Support/browsh/firefox_profile","moz:useNonSpecCompliantPointerOrigin":false,"moz:webdriverClick":true}}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 25:[1,1,null,{"value":null}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 25:[1,2,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 25:[1,3,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 25:[1,4,null,{"value":null}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 25:[1,5,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 25:[1,6,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 25:[1,7,null,{"value":null}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 25:[1,8,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 25:[1,9,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 26:[1,10,null,{"value":null}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 26:[1,11,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 26:[1,12,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 26:[1,13,null,{"value":null}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 26:[1,14,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 26:[1,15,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 26:[1,16,null,{"value":null}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 26:[1,17,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 26:[1,18,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 26:[1,19,null,{"value":null}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 26:[1,20,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 26:[1,21,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 26:[1,22,null,{"value":null}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 26:[1,23,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 26:[1,24,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 26:[1,25,null,{"value":null}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 26:[1,26,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 26:[1,27,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 26:[1,28,null,{"value":null}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 26:[1,29,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 26:[1,30,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 26:[1,31,null,{"value":null}]
Sending `executeScript` to Firefox Marionette
FF-MRNT: 26:[1,32,null,{"value":null}]
Sending `setContext` to Firefox Marionette
FF-MRNT: 26:[1,33,null,{"value":null}]
Sending `addon:install` to Firefox Marionette
Starting Firefox in headless mode
Using Firefox at: /Applications/Firefox.app/Contents/MacOS/firefox
FF-MRNT: 62:[1,34,null,{"value":"{8ff2d753-2dc8-46de-a837-fa28331d9fcf}"}]
Starting Browsh CLI client
Using default profile at: /Users/zhengdelweng/Library/Application Support/browsh/firefox_profile
Webextension not connected. Message not sent: /tty_size,334,75
```
  * Include your OS and terminal name
macOS 10.13.6; browsh 1.4.12

**Comment by tombh** [comment]
I think this is the case of Firefox not being able to access https://brow.sh, whether because of slow internet, or being behind a proxy/firewall.

**Comment by tobimensch** [comment]
@tombh 
Maybe browsh could always start it's own local webserver for serving its own documentation. We need it anyways to forward local documents to browsh, which is issue #71 .
The advantage here being, that people at least will understand, that the bug is due to a problem with their internet connection, which makes it less likely for them to repost this bug.
So browsh would always start a server on say localhost:3422 and visit localhost:3422/browsh/documentation by default.

**Comment by tombh** [comment]
Yep, you're absolutely right, it's gotta be done this way hasn't it. You gotta be able to start Browsh without error if there isn't any internet.

**Comment by tombh** [comment]
Closing this to be addressed in #194
